### ISSUE TITLE
influxd restore panic

### ISSUE THREAD

**Issue by bolek2000** [influxdata/influxdb#9701]
### Bug report

__System info:__ 
InfluxDB 1.5.1, Ubuntu 16.04

__Steps to reproduce:__

1. Trying to restore a full backup made on another machine (size around 31 GB)
2. influxd restore -db telegraf -newdb restore_telegraf -portable -host '127.0.0.1:8089' -rp autogen /root/IDB_Backups/telegraf/ 

__Expected behavior:__ 
Restoring data to InfluxDB

__Actual behavior:__
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x96fac6]

goroutine 1 [running]:
github.com/influxdata/influxdb/cmd/influxd/restore.(*Command).updateMetaPortable(0xc4201f4000, 0x0, 0xf0)
	/go/src/github.com/influxdata/influxdb/cmd/influxd/restore/restore.go:316 +0x86
github.com/influxdata/influxdb/cmd/influxd/restore.(*Command).runOnlinePortable(0xc4201f4000, 0xc4200100c0, 0x8)
	/go/src/github.com/influxdata/influxdb/cmd/influxd/restore/restore.go:104 +0x2f
github.com/influxdata/influxdb/cmd/influxd/restore.(*Command).Run(0xc4201f4000, 0xc4200100c0, 0x8, 0x8, 0xc42007a200, 0xc42007a2a0)
	/go/src/github.com/influxdata/influxdb/cmd/influxd/restore/restore.go:78 +0x209
main.(*Main).Run(0xc42005bf40, 0xc4200100b0, 0x8, 0x8, 0x5acc8de7, 0x3579b4ee)
	/go/src/github.com/influxdata/influxdb/cmd/influxd/main.go:115 +0xf04
main.main()
	/go/src/github.com/influxdata/influxdb/cmd/influxd/main.go:45 +0x16a

__Additional info:__ 
When I opened the bug I thought the cause of the error is, that the InfluxDB service is not running, but I started it, can connect to the DB, but still get this error. 

netstat -plunt shows:
tcp        0      0 127.0.0.1:8089          0.0.0.0:*               LISTEN      10341/influxd 

Can you give hints how to debug this ?

**Comment by bolek2000** [comment]
The Backup was created on a machine with InfluxDB 1.5.1, ubuntu 16.04 without the `-portable` option. So if I understand correctly, I have a backup in legacy format. 
So if I remove `-portable` then I get normal error messages:
```
influxd restore -datadir /var/lib/influxdb/data -db telegraf -newdb restore_telegraf -host '127.0.0.1:8089' -rp autogen /root/IDB_Backups/telegraf/ 
2018/04/10 12:58:47 Restoring offline from backup /root/IDB_Backups/telegraf/restore_telegraf.*
restore: no backup files for /root/IDB_Backups/telegraf/restore_telegraf.* in /root/IDB_Backups/telegraf/
```
But still I cannot restore the backup due to a new error message. Do I need to rename all the files to `restore_telegraf.*` ??

**Comment by bolek2000** [comment]
So to see what happens if I renamed all the files (except `meta.00`) with the prefix `restore_` so the files have the same name as my new DB I want to create. Afterwards the restore ran smoothly, but still my DB was not visible, so I ran `systemctl restart influxdb` an get:
influxd[15276]: run: open server: open tsdb store: mkdir /var/lib/influxdb/data/restore_telegraf/_series: permission denied
After `chown -R influxdb:influxdb /var/lib/influxdb/data/restore_telegraf/` the service started and opened the shards (I can see in syslog). But in the influx client I cannot the the DB if I type:
```
> show databases
name: databases
name
----
_internal
telegraf
graphite
```
I'm lost at the moment ?

**Comment by dgnorton** [comment]
@bolek2000 is there a `.manifest` file in `/root/IDB_Backups/telegraf/`?

**Comment by dgnorton** [comment]
Troubleshooting notes:

Looks like [`cmd.manifestMeta` can be `nil` here](https://github.com/influxdata/influxdb/blob/v1.5.1/cmd/influxd/backup_util/backup_util.go#L156) and [referenced here](https://github.com/influxdata/influxdb/blob/v1.5.1/cmd/influxd/restore/restore.go#L316).

**Comment by bolek2000** [comment]
No .manifest existing. The backup was created per database with -database option. It seems that I can only use the old legacy restore procedure (with data loss) described in the docs. -online also doensn't work. The only thing I didn't explicitly do is restoring only the metastore. But I guess the old meta.00 file will only know the old databases, not the new one I want to create, isn't it ?

**Comment by bolek2000** [comment]
So this works for me to use the restored data (with data loss), but I am restoring on a test instance. The backup files have the original names from backup:
```
# systemctl stop influxdb
# influxd restore -metadir /var/lib/influxdb/meta /root/IDB_Backups/telegraf/
Using metastore snapshot: telegraf/meta.00
# influxd restore -datadir /var/lib/influxdb/data -db telegraf -newdb telegraf -host '127.0.0.1:8089' -rp autogen /root/IDB_Backups/telegraf/
2018/04/10 14:38:25 Restoring offline from backup /root/IDB_Backups/telegraf/telegraf.*
# chown -R influxdb:influxdb /var/lib/influxdb
# systemctl start influxdb
```

So I am leaving the panic message for you to fix. Thanks so far....

**Comment by aanthony1243** [comment]
@bolek2000   This bug is fixed and will go out on the next release: #9634   but there is a workaround for 1.5.1: 

This issue occurs because you did not use -portable when you create the backup, and so the 'legacy' file format was generated.  You can still use the new features in restore!  Instead of   -portable,   try  -online,   which behaves the same as -portable but consumes the old legacy format including the meta.00 and related backup files. 

Even after the bugfix, we will avoid the panic but instead the restore would have printed an error message and exited without restoring anything.  Your final solution then will be to use -online  if you have generated the legacy backup format.
